test/fuzz/transformation_changing_memory_semantics_test.cpp

```diff
@@ -0,0 +1,479 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "source/fuzz/transformation_changing_memory_semantics.h"
+
+#include "gtest/gtest.h"
+#include "source/fuzz/fuzzer_util.h"
+#include "source/fuzz/instruction_descriptor.h"
+#include "test/fuzz/fuzz_test_util.h"
+
+namespace spvtools {
+namespace fuzz {
+namespace {
+
+TEST(TransformationChangingMemorySemanticsTest, NotApplicable) {
+  const std::string shader = R"(
+               OpCapability Shader
+               OpCapability Int8
+          %1 = OpExtInstImport "GLSL.std.450"
+               OpMemoryModel Logical GLSL450
+               OpEntryPoint Fragment %4 "main"
+               OpExecutionMode %4 OriginUpperLeft
+               OpSource ESSL 320
+          %2 = OpTypeVoid
+          %3 = OpTypeFunction %2
+          %6 = OpTypeInt 32 1
+          %7 = OpTypeInt 8 1
+          %9 = OpTypeInt 32 0
+         %26 = OpTypeFloat 32
+          %8 = OpTypeStruct %6
+         %10 = OpTypePointer StorageBuffer %8
+         %11 = OpVariable %10 StorageBuffer
+         %19 = OpConstant %26 0
+         %18 = OpConstant %9 1
+         %12 = OpConstant %6 0
+         %13 = OpTypePointer StorageBuffer %6
+         %15 = OpConstant %6 4
+         %16 = OpConstant %6 7
+         %17 = OpConstant %7 4
+         %20 = OpConstant %6 80 ; SequentiallyConsistent | UniformMemory
+         %97 = OpConstant %6 64 ; None | UniformMemory
+         %98 = OpConstant %6 66 ; Acquire | UniformMemory
+         %99 = OpConstant %6 68 ; Release | UniformMemory
+        %100 = OpConstant %6 72 ; AcquireRelease | UniformMemory
+          %4 = OpFunction %2 None %3
+          %5 = OpLabel
+         %14 = OpAccessChain %13 %11 %12
+         %21 = OpAtomicLoad %6 %14 %15 %20
+         %23 = OpAtomicExchange %6 %14 %15 %98 %16
+         %25 = OpAtomicCompareExchange %6 %14 %15 %20 %20 %16 %15
+         %24 = OpAccessChain %13 %11 %12
+               OpAtomicStore %14 %15 %99 %16
+               OpReturn
+               OpFunctionEnd
+  )";
+  const auto env = SPV_ENV_UNIVERSAL_1_3;
+  const auto consumer = nullptr;
+  const auto context = BuildModule(env, consumer, shader, kFuzzAssembleOption);
+  spvtools::ValidatorOptions validator_options;
+  ASSERT_TRUE(fuzzerutil::IsValidAndWellFormed(context.get(), validator_options,
+                                               kConsoleMessageConsumer));
+  TransformationContext transformation_context(
+      MakeUnique<FactManager>(context.get()), validator_options);
+
+  // Bad: Instruction does not exist.
+  ASSERT_FALSE(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(150, SpvOpAtomicLoad, 0), 0, 100)
+                   .IsApplicable(context.get(), transformation_context));
+
+  // Bad: Operand index does not equal 2 or 3.
```

Comment:
It looks like maybe your tests need to be updated? We said the operand index will be 0 or 1.

Reply:
Updated.